rfd/0194-k8s-rbac-simplification.md

```diff
@@ -0,0 +1,220 @@
+editing, go back to YAML editor.
+```
+
+The `kubernetes_resources` section currently "hidden" behind a button should be
+open by default.
+
+Alternatively, we could keep `kubernetes_group` under an "advanced" button,
+which would show what we see today by default as well as a link to a specific
+documentation page.
+
+#### CLI
+
+Using `tctl`, the fields will only be visible if they are set. Existing roles
+will have the existing values, even if already set to `kubernetes_master`,
+however, new roles can be created without the field and will be surfaced back
+without the field either.
+
+#### Direct enrollment
+
+When enrolling a cluster "directly" (i.e. by adding the `kubernetes_service`
+section in the `teleport.yaml` config), the initial setup of roles/rolebindings
+still needs to be performed calling the following script:
+[get-kubeconfig.sh](../examples/k8s-auth/get-kubeconfig.sh).
+
+It would be interesting to add a field to the config to allow specifying the
+default group that will be used for impersonation for advanced usecases to
+allow admins who prefer to rely on _Kubernetes_ RBAC instead of letting the
+agent be authoritative.
+
+#### Helm enrollment
+
+When using the Helm chart, nothing is expected to change, the chart sets up the
+service account for the agent, as well as the role/binding for impersonation.
+
+As _Helm_ is the favored way to get up and running, we may consider creating a
+dedicated group `teleport:master` or similar instead of using `system:master`.
+This would be cleaner but would rely on the Helm chart which would fracture the
+customer base and make support more difficult.
+
+It would be interesting to add an argument to the chart to allow specifying
+the default group that will be used for impersonation for advanced usecases.
+
+#### Resource enrollment UI
+
+On the Web UI, the initial page generates a _Helm_ commandline.
+
+While possible via the _Helm_ chart, as it is discouraged and for advanced uses
+only, we should't provide a UI to set a specific group to replace
+`system:masters`.
+
+After enrollment, a test page is shown, prompting for a `kubernetes_group`
+value, which defaults to the user's trait.
+With the proposed changes, that page would be skipped, using the default
+`system:masters` or the configured default for the cluster.
+
+#### Cluster discovery
+
+Similar to the _Helm_ enrollment, the discovery service sets up the
+role/rolebindings for impersonation.
+It would be interesting to add a field to the `discovery_service` section in
+the configuration to allow setting a different group rather than
+`system:masters` for the newly discovered clusters.
+
+### Backward/Forward compatibility
+
+The idea is to keep the impersonation and "hide" `kubernetes_group` and
+`kubernetes_users`, existing setups will keep working, with the downside that
+the web role editor may be disabled.
+
+As there is a (tiny) risk that the new behavior would unexpectedly grant access
```

Comment:
This will break our compatibility guarantees and we can't merge it as described here.

My suggestion is to:
- grant `cluster-admin` to Teleport
- introduce role v8
- if user roles have <=v7, we use the old method
- if roles are >=v8, we don't allow setting kube groups and users and we force people to use kubernetes_resources. This method will only work if all roles are v8 or up